src/physfs_tar.h

```diff
@@ -0,0 +1,299 @@
+	DIRTYPE          =   '5', /* directory */
+	FIFOTYPE         =   '6', /* FIFO special */
+	CONTTYPE         =   '7', /* reserved */
+	XHDTYPE          =   'x', /* Extended header referring to the next file in the archive */
+	XGLTYPE          =   'g', /* Global extended header */
+	GNUTYPE_DUMPDIR	 =   'D', /* names of files been in the dir at the time of the dump */
+	GNUTYPE_LONGLINK =   'K', /* next file has long linkname.  */
+	GNUTYPE_LONGNAME =   'L', /* next file has long name.  */
+	GNUTYPE_MULTIVOL =   'M', /* continuation file started on another volume.  */
+	GNUTYPE_SPARSE   =   'S', /* This is for sparse files.  */
+	GNUTYPE_VOLHDR   =   'V', /* This file is a tape/volume header.  Ignore it on extraction.  */
+	SOLARIS_XHDTYPE  =   'X', /* Solaris extended header */
+};
+
+enum mode {
+	TOEXEC           = 1 << 0, /* execute/search by other */
+	TOWRITE          = 1 << 1, /* write by other */
+	TOREAD           = 1 << 2, /* read by other */
+	TGEXEC           = 1 << 3, /* execute/search by group */
+	TGWRITE          = 1 << 4, /* write by group */
+	TGREAD           = 1 << 5, /* read by group */
+	TUEXEC           = 1 << 6, /* execute/search by owner */
+	TUWRITE          = 1 << 7, /* write by owner */
+	TUREAD           = 1 << 8, /* read by owner */
+	TSVTX            = 1 << 9, /* reserved */
+	TSGID            = 1 << 10, /* set GID on execution */
+	TSUID            = 1 << 11, /* set UID on execution */
+};
+
+/* POSIX header.  */
+struct posix_header
+{				  /* byte offset */
+	char name[100];           /*   0 */
+	char mode[8];             /* 100 */
+	char uid[8];              /* 108 */
+	char gid[8];              /* 116 */
+	char size[12];            /* 124 */
+	char mtime[12];           /* 136 */
+	char chksum[8];           /* 148 */
+	char typeflag;            /* 156 */
+	char linkname[100];       /* 157 */
+	char magic[TMAGLEN];      /* 257 */
+	char version[2];          /* 263 */
+	char uname[32];           /* 265 */
+	char gname[32];           /* 297 */
+	char devmajor[8];         /* 329 */
+	char devminor[8];         /* 337 */
+	char prefix[155];         /* 345 */
+	                          /* 500 */
+};
+
+/* tar Header Block, from POSIX 1003.1-1990.  */
+union block
+{
+  char buffer[BLOCKSIZE];
+  struct posix_header header;
+};
+
+static PHYSFS_uint64 TAR_decodeOctal(char* data, size_t size) {
+    unsigned char* currentPtr = (unsigned char*) data + size;
+    PHYSFS_uint64 sum = 0;
+    PHYSFS_uint64 currentMultiplier = 1;
+    unsigned char* checkPtr = currentPtr;
+
+    for (; checkPtr >= (unsigned char*) data; checkPtr--) {
+        if ((*checkPtr) == 0 || (*checkPtr) == ' ') {
+            currentPtr = checkPtr - 1;
+        }
+    }
+    for (; currentPtr >= (unsigned char*) data; currentPtr--) {
+        sum += (*currentPtr - 48) * currentMultiplier;
+        currentMultiplier *= 8;
+    }
+    return sum;
+}
+
+static bool TAR_encodeOctal(char* data, size_t size, PHYSFS_uint64 i) {
```

Comment:
This isn't actually used: Only referenced by the mode functions which themselves aren't used. can be commented out (also see below)